File: domains/query-execution/steps/route.py
"""S2: route by QueryIntent.query_path."""

from __future__ import annotations


def run(ctx):
    query_path = ctx.get("query_path") or ctx.get("query_intent", {}).get("query_path")
    if query_path not in {"dashboard", "raw_analysis"}:
        return StepOutcome.fail(f"unsupported query_path: {query_path}")
    return StepOutcome.next({"query_path": query_path}, branch=query_path)

File: domains/query-execution/steps/route.ts
/** S2: route by QueryIntent.query_path. */

import { StepOutcome } from "../scheduler/scheduler.js";

export function run(ctx: Record<string, any>): StepOutcome {
  const query_path = ctx["query_path"] || (ctx["query_intent"] ?? {})["query_path"];
  if (query_path !== "dashboard" && query_path !== "raw_analysis") {
    return StepOutcome.fail(`unsupported query_path: ${query_path}`);
  }
  return StepOutcome.next({ query_path }, query_path);
}
